src/snowflake/snowpark/files.py

```diff
@@ -50,19 +53,24 @@ def open(
         file_location: str,
         mode: str = "r",
         is_owner_file: bool = False,
+        *,
+        require_scoped_url: bool = True,
     ) -> SnowflakeFile:
         """
         Returns a :class:`~snowflake.snowpark.file.SnowflakeFile`.
         In UDF and Stored Procedures, the object works like a Python IOBase object and as a wrapper for an IO stream of remote files. The IO Stream is to support the file operations defined in this class.
 
-	All files are accessed in the context of the UDF owner and therefore when accessing call files user should prefer setting is_owner_file to default false value as it only allows scoped URLs and therefore a caller cannot inadvertently or maliciously access the owner's files.
+	All files are accessed in the context of the UDF owner and therefore when accessing caller files the user should prefer setting require_scoped_url to True (default value) as it requires the use of a scoped URL and therefore a caller cannot inadvertently or maliciously access the owner's files.
```

Comment:
We might want to explain a bit more here. We did a few paragraphs on the java one.
https://github.com/snowflakedb/thundersnow/pull/979/files